src/app_feup/lib/controller/library/library.dart

```diff
@@ -1,49 +1,271 @@
 import 'dart:async';
+import 'dart:io';
 
-import 'package:html/parser.dart';
-import 'package:http/http.dart';
+import 'package:html/dom.dart';
+import 'package:html/parser.dart' as html;
 import 'package:logger/logger.dart';
+import 'package:tuple/tuple.dart';
 import 'package:uni/controller/library/library_interface.dart';
+import 'package:uni/controller/library/library_utils.dart';
 import 'package:uni/controller/library/parser_library.dart';
 import 'package:uni/controller/library/parser_library_interface.dart';
+import 'package:uni/controller/local_storage/app_shared_preferences.dart';
 import 'package:uni/model/entities/book.dart';
 import 'package:http/http.dart' as http;
 
+import 'package:cookie_jar/cookie_jar.dart';
+
 extension UriString on String {
   /// Converts a [String] to an [Uri].
   Uri toUri() => Uri.parse(this);
 }
 
-final String baseUrl = 'https://catalogo.up.pt/F';
+class Library implements LibraryInterface {
+  final _client = http.Client();
+  final cookies = CookieJar();
+  Cookie pdsCookie;
+  String _username;
+  String _password;
 
-// TODO receive this WRD from the search filter
-String baseSearchUrl(String query) =>
-    'https://catalogo.up.pt/F/?func=find-b&request=$query&find_code=WRD';
+  // User can have multiple faculties, should we search for books in all?
```

Comment:
```suggestion
  // TODO Multiple faculties in the library catalog
```
This doesn't seem worth it right now. I think we should just leave it for the far future